Enable Windowing for Manual Image Fusion

mattbrgn opened the pull request:
Additions:
- Fixed windowing capability for manual image fusion
- Adjusted buttons in translate/rotate menu for better styling
- Added comments and docstrings for readability

NOTE: Will has left a TODO to fix VTK Engine to work natively with the values OnkoDICOM uses for other views, we couldn't get it to work (produced only very dense bone), so this is a workaround.

## Summary by Sourcery

Enable and refine windowing support for manual image fusion: integrate window/level controls into fusion views, modularize fusion tab setup in MainPage, improve manual load and transform handling, and polish TranslateRotateMenu styling.

New Features:
- Propagate window and level changes to manual fusion views via the windowing slider and model
- Extend ManualFusionLoader to emit loaded VTK engine images, store fusion images in PatientDictContainer, and initialize window/level settings
- Enhance TranslateRotateMenu with dynamic icon updates, button styling, and integrated window/level callbacks

Bug Fixes:
- Fix windowing capability for manual image fusion

Enhancements:
- Refactor MainPage.create_image_fusion_tab into modular helper methods to improve code organization
- Add docstrings and logging improvements across MainPage, TranslateRotateMenu, VTKEngine, and loading logic
- Adjust margins and icon handling for translate/rotate/interrogation buttons in the fusion menu
- Enhance Windowing model to use custom presets for manual fusion and coordinate slider updates

Documentation:
- Add comments and docstrings throughout the new and refactored methods to improve readability

mattbrgn committed:
Merge branch 'master' into dev

Nowhere4Nothing committed:
Update with Sourceys advice

Made update colour overlay a utility function so there is no duplicate code. Both now call it

Re factored translate rotation menu to have helper functions for readbility

added spacing so the buttons dont touch

mattbrgn committed:
Fixed Mouse Interaction being Blocked

- Metadata labels were blocking mouse clicks

Nowhere4Nothing committed:
Fixed bug Reopen

Fixed the bug where the program would crash when opening a new patient after opening the image fusion tab.

added a method in main page create image fusion tab to clean up old widgets

this bug was because certain widgets and sliders were not getting deleted properly and the new patient would reference the old references crashing the program

Nowhere4Nothing committed:
Updated as per sourcerys advice

MainPage.py now contains many _init_... helper methods – it might be cleaner to move them into a dedicated ImageFusionTabBuilder (or similar) to improve cohesion and reduce file complexity.

added a new class that takes a lot of the code away and makes it more readable.

Kahreiru reviewed (approved):
i'm not seeing issues.

sjswerdloff reviewed (approved):
Good enough, but I'd prefer that the logging issues I mentioned get addressed as well as  explaining what interrogate means in the context it is being used.
Someone will need to maintain the code, and not knowing what something is or means that is buried deep in the code would make that more difficult.

Nowhere4Nothing commented on src/View/ImageFusion/BaseViewerGUI.py:
```diff
@@ -340,6 +342,19 @@ def update_overlay_offset(self, offset, orientation=None, slice_idx=None):
+        """
+                Handles changes to the mouse interaction mode for the fusion view.
+
+                This method updates the internal mouse mode state and manages the interrogation window position.
```
its the nickname that Andrew calls the window button. The one that has a window you can move that only shows the base layer inside and shows the overlayer and base outside. Ill explain it better in the doc

Nowhere4Nothing replied:
added

Nowhere4Nothing committed:
Update VTK, Windowin & Baseviewer

Updated these files to add Stuarts tips

Changed a logging error to a warning
Added consistent variables to VTK
swapped the window and level in the call to vtk in windowing
added a note in the docstring in base fusion view to explain interrogation window

Nowhere4Nothing commented on src/Model/Windowing.py:
```diff
@@ -16,8 +16,27 @@ def windowing_model(text, init):
+            windowing_limits = dict_windowing_moving[text]
+        else:
+            # Fallback to a hardcoded set or the fixed image's presets
+            custom_presets = {
```
Swapped them

Nowhere4Nothing committed:
Missed one point by Stuart

Moved the import files each to their own line

added another line in the doc string explaining the interrogation window
